jme3-core/src/com/jme3/gde/core/assets/actions/MergeAnimationsAction.java

```diff
@@ -0,0 +1,208 @@
+            final AnimComposer sourceAnimComposer = findAnimComposer(sourceSpatial, null);
+            if (sourceAnimComposer == null) {
+                DialogDisplayer.getDefault().notify(new NotifyDescriptor.Message(
+                            String.format("%s has no AnimComposer.", sourceSpatial),
+                            NotifyDescriptor.ERROR_MESSAGE));
+                return;
+            }
+            copyClips(sourceAnimComposer, targetAnimComposer, targetAnimComposerSpatial.getControl(SkinningControl.class).getArmature());
+        }
+        logger.log(Logger.Level.INFO, "Merging animations done. Saving.");
+        try {
+            targetAsset.saveAsset();
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+    }
+    
+    private static AnimClip retargetClip(
+            AnimClip clip, Armature armature, String clipName) {
+        AnimTrack[] animTracks = clip.getTracks();
+        AnimClip result = new AnimClip(clipName);
+        for (AnimTrack animTrack : animTracks) {
+            if (MyAnimation.isJointTrack(animTrack)) {
+                TransformTrack transformTrack = (TransformTrack) animTrack;
+                HasLocalTransform animTarget = transformTrack.getTarget();
+                Joint animJoint = (Joint) animTarget;
+                String jointName = animJoint.getName();
+                Joint charaJoint = armature.getJoint(jointName);
+                if (charaJoint != null) {
+                    transformTrack.setTarget(charaJoint);
+                    AnimationEdit.addTrack(result, transformTrack);
+                }
+            }
+        }
+        return result;
+    }
+
+    
+    private void copyClips(final AnimComposer from, final AnimComposer to, Armature toArmature) {
+        final Collection<AnimClip> animClips = from.getAnimClips();
+        for(AnimClip animClip: animClips) {
+            to.addAnimClip(retargetClip(animClip, toArmature, animClip.getName()));
+            logger.log(Logger.Level.DEBUG, String.format("Added anim clip %s", animClip.getName()));
+        }
+    }
+
+    private Confirmation createSelector() {
+        final Object[] spatials = new Object[this.spatials.size() + 1];
+        int index = 0;
+        for (Iterator<SpatialAssetDataObject> it = this.spatials.iterator(); it.hasNext();) {
+            final SpatialAssetDataObject spatialAssetDataObject = it.next();
+            spatials[index++] = spatialAssetDataObject.getName();
+        }
+        spatials[index++] = CANCEL;
+        final NotifyDescriptor.Confirmation message =
+                new NotifyDescriptor.Confirmation(
+                        "Select spatial to copy animations to.");
+        message.setOptions(spatials);
+        
+        DialogDisplayer.getDefault().notify(message);
+        
+        return message;
+    }
+    
+    private AnimComposer findAnimComposer(Spatial spatial, AnimComposer animComposer) {
```

Comment:
You can rewrite this method in a much simpler way like this:

Usage: `AnimComposer animComposer = findControl(spatial, AnimComposer.class);`

```java
private static <T extends Control> T findControl(Spatial sp, Class<T> clazz) {
        T control = sp.getControl(clazz);
        if (control != null) {
            return control;
        }
        if (sp instanceof Node) {
            for (Spatial child : ((Node) sp).getChildren()) {
                control = findControl(child, clazz);
                if (control != null) {
                    return control;
                }
            }
        }
        return null;
    }
```

Reply:
I'll save this for later. It might be a good method to have in a central place.